### PR TITLE
Benchmarks haml hamlit slim

### PR DESCRIPTION
**Pull request by konung**
Hi @joeldrapper 

We briefly touched on benchmarks in this discussion https://github.com/joeldrapper/phlex/discussions/443 

This is a first stab at benchmarks, just based of what you already had.  Obviously Phlex code is not compiled, as you said before in the discussion and on Remote Ruby podcast that I just listened to, it still needs work.  My expectations, is that once  it passes through compiler, it should be on par with others. 

To my surprise Slim was faster, than Haml & Hamlt , but I'm guessing that's because the test is trivial.  I'm quite possibly doing something wrong as well 😸 

Hopefully this is a start in the right direction. 

My results running on ruby 3.2  `ruby --jit bench.rb`

```shell
ruby 3.2.0 (2022-12-25 revision a528908271) +YJIT [x86_64-darwin22]
Warming up --------------------------------------
          Phlex Page     5.873k i/100ms
           HAML Page    12.048k i/100ms
         Hamlit Page    11.044k i/100ms
           Slim Page    19.863k i/100ms
Calculating -------------------------------------
          Phlex Page     62.061k (± 4.4%) i/s -    311.269k in   5.025985s
           HAML Page    138.166k (± 4.3%) i/s -    698.784k in   5.067378s
         Hamlit Page    138.280k (± 4.9%) i/s -    695.772k in   5.045130s
           Slim Page    197.794k (± 3.2%) i/s -    993.150k in   5.026305s

Comparison:
           Slim Page:   197794.2 i/s
         Hamlit Page:   138279.8 i/s - 1.43x  (± 0.00) slower
           HAML Page:   138165.6 i/s - 1.43x  (± 0.00) slower
          Phlex Page:    62061.1 i/s - 3.19x  (± 0.00) slower

```

### CODE REVIEW

**Comment by konung** [comment]
Also to clarify - the bench is really setup to only compare "rendering" of html, the assumption being that in production code, all templates will already be preloaded.

**Comment by konung** [comment]
After a bit of thought - running a simple benchmark, that produce static page is not bery usful in most application. Even systems like Jekyll that produce static content take input for templates and "partials"

Updating benchmark to do something useful - like showing a table of products, with some styling and dynamic attributes. 

Now numbers are actually on the same playing field: 😆 

```shell
ruby 3.2.0 (2022-12-25 revision a528908271) +YJIT [x86_64-darwin22]
Warming up --------------------------------------
          Phlex Page     2.985k i/100ms
           HAML Page     3.526k i/100ms
         Hamlit Page     3.518k i/100ms
           Slim Page     4.144k i/100ms
Calculating -------------------------------------
          Phlex Page     29.962k (± 3.4%) i/s -    152.235k in   5.087114s
           HAML Page     35.237k (± 3.6%) i/s -    176.300k in   5.011065s
         Hamlit Page     35.477k (± 1.5%) i/s -    179.418k in   5.058469s
           Slim Page     41.240k (± 3.6%) i/s -    207.200k in   5.031416s

Comparison:
           Slim Page:    41239.9 i/s
         Hamlit Page:    35476.7 i/s - 1.16x  (± 0.00) slower
           HAML Page:    35237.2 i/s - 1.17x  (± 0.00) slower
          Phlex Page:    29962.2 i/s - 1.38x  (± 0.00) slower
```

File: README.md [hunk 1]
```diff
@@ -6,6 +6,14 @@ Phlex lets you compose web views in pure Ruby. It’s super-fast, thread-safe an
...
 Documentation can be found at [www.phlex.fun](https://www.phlex.fun).
 
+### Testing
+
+We use [`sus`](https://github.com/ioquatix/sus) execute tests. To run gem's testsuite use the following command
+
+```
+bundle exec sus
+```
+
```
Comment (joeldrapper) [inline README.md:16]: This is in `CONTRIBUTING.md`. I wonder if we should just link that from the README under a "Contributing" heading.

Comment (konung) [reply README.md:16]: Ah. My bad missed it.  I just skimmed Readme, and didn't see anything.  :)

**Comment by joeldrapper** [comment]
@konung I really appreciate you doing these. 🙏 I will probably copy them into a new dedicated benchmarking repo at some point but for now I’m going to close this PR.

**Comment by konung** [comment]
No problem . Whatever works 😁

Have a great day!

Nick

**Pull request closed without merging**